From 54386cf9f4143c4d3b3445b191608d0a53376ca4 Mon Sep 17 00:00:00 2001
From: Douile <25043847+Douile@users.noreply.github.com>
Date: Fri, 9 Dec 2022 22:23:34 +0000
Subject: [PATCH 1/2] Re-enter passphrase on fail

---
 grub-core/disk/cryptodisk.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/grub-core/disk/cryptodisk.c b/grub-core/disk/cryptodisk.c
index 14308822a..39446b84a 100644
--- a/grub-core/disk/cryptodisk.c
+++ b/grub-core/disk/cryptodisk.c
@@ -1102,7 +1102,8 @@ grub_cryptodisk_scan_device_real (const char *name,
       {
 	/* Get the passphrase from the user, if no key data. */
 	askpass = 1;
-	part = grub_partition_get_name (source->partition);
+    enter_passphrase:
+	  part = grub_partition_get_name (source->partition);
 	grub_printf_ (N_("Enter passphrase for %s%s%s (%s): "), source->name,
 		     source->partition != NULL ? "," : "",
 		     part != NULL ? part : N_("UNKNOWN"),
@@ -1115,13 +1116,19 @@ grub_cryptodisk_scan_device_real (const char *name,
 
 	if (!grub_password_get ((char *) cargs->key_data, GRUB_CRYPTODISK_MAX_PASSPHRASE))
 	  {
-	    grub_error (GRUB_ERR_BAD_ARGUMENT, "passphrase not supplied");
-	    goto error;
+      grub_printf_ ("You must supply a passphrase\n");
+      goto enter_passphrase;
+	    //grub_error (GRUB_ERR_BAD_ARGUMENT, "passphrase not supplied");
+	    //goto error;
 	  }
 	cargs->key_len = grub_strlen ((char *) cargs->key_data);
       }
 
     ret = cr->recover_key (source, dev, cargs);
+    if (ret == GRUB_ACCESS_DENIED) {
+      grub_printf_ ("Invalid passphrase\n");
+      goto enter_passphrase;
+    }
     if (ret != GRUB_ERR_NONE)
       goto error;
 
-- 
2.38.1

