From 3a4dde2297e93beb1af4641516361317cce092fd Mon Sep 17 00:00:00 2001
From: Douile <25043847+Douile@users.noreply.github.com>
Date: Sun, 6 Mar 2022 23:04:14 +0000
Subject: [PATCH 2/2] Add custom heading

---
 grub-core/kern/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/grub-core/kern/main.c b/grub-core/kern/main.c
index 73967e2f5..c69f92f63 100644
--- a/grub-core/kern/main.c
+++ b/grub-core/kern/main.c
@@ -272,8 +272,9 @@ grub_main (void)
 
   /* Hello.  */
   grub_setcolorstate (GRUB_TERM_COLOR_HIGHLIGHT);
-  grub_printf ("Welcome to GRUB!\n\n");
+  grub_printf ("Welcome to GRUB (Tom edition)!\n\n");
   grub_setcolorstate (GRUB_TERM_COLOR_STANDARD);
+  grub_printf ("You better not be here to hek me\n\n");
 
   /* Init verifiers API. */
   grub_verifiers_init ();
-- 
2.35.1

